docs/TYPES_VERSION_STRATEGY.md

```diff
@@ -0,0 +1,119 @@
+# TypeScript Definitions Version Strategy
+
+This document describes how create-p5 handles TypeScript type definitions for different p5.js versions.
+
+## The Problem
+
+The @types/p5 package uses its own versioning scheme (latest: 1.7.7) that doesn't directly match p5.js versions (latest: 2.1.1). Additionally, p5.js started bundling its own TypeScript definitions starting from version 2.0.2.
+
+## The Solution
+
+We use a two-tier strategy based on the p5.js version:
+
+### For p5.js 1.x (versions 1.0.0 - 1.11.x)
+
+**Use @types/p5 package from DefinitelyTyped**
+
+- Source: `https://cdn.jsdelivr.net/npm/@types/p5@<version>/`
+- Latest available: `@types/p5@1.7.7`
+- Files needed:
```

Comment:
On further investigation, I think these files aren't enough.  [Detail here](https://discord.com/channels/820953351817789480/1445092778294902795/1445428855484448780)

Reply:
Thanks for your feedback! What would you advise? Should `create-p5js` download all the files linked in `index.d.ts` like `<reference path="./src/accessibility/describe.d.ts" />` etc? Or maybe download them and use a bundler to combine them into one file?